cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,307 @@
+            entries.put(HEALTHCHECK_USERNAME, new UserEntry(HEALTHCHECK_USERNAME, passwordGenerator.generate(), ApiCredentials.Role.USER));
+        }
+
+        return entries;
+    }
+
+    /**
+     * Creates map with API usernames, passwords, and credentials file for Strimzi-managed API users secret.
+     *
+     * @param entries List of API user entries containing API user credentials
+     *
+     * @return Map containing Cruise Control API auth credentials
+     */
+    public static Map<String, String> generateMapWithApiCredentials(Map<String, UserEntry> entries) {
+        Map<String, String> data = new HashMap<>(3);
+        data.put(REBALANCE_OPERATOR_PASSWORD_KEY, Util.encodeToBase64(entries.get(REBALANCE_OPERATOR_USERNAME).password()));
+        data.put(HEALTHCHECK_PASSWORD_KEY, Util.encodeToBase64(entries.get(HEALTHCHECK_USERNAME).password()));
+        data.put(AUTH_FILE_KEY, Util.encodeToBase64(generateApiAuthFileAsString(entries)));
+        return data;
+    }
+
+    /**
+     * Returns Cruise Control API auth credentials file following Jetty's
+     * HashLoginService's file format: username: password [,rolename ...]
+     * as a String.
+     *
+     * @param entries Map of API user entries containing API user credentials.
+     *
+     * @return Returns Cruise Control API auth credentials file as a String.
+     */
+    private static String generateApiAuthFileAsString(Map<String, UserEntry> entries) {
+        StringBuilder sb = new StringBuilder();
+        for (UserEntry e : entries.values()) {
+            sb.append(e.username).append(": ").append(e.password).append(", ").append(e.role).append("\n");
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Generates a new API secret for Cruise Control by aggregating credentials from various sources.
+     * This method collects API credentials from three potential sources:
+     *   (1) Old Cruise Control API secret
+     *   (2) User-managed API secret
+     *   (3) Topic operator-managed API secret.
+     * It uses these credentials to create a comprehensive map of API credentials, which is then used to generate a new API secret
+     * for Cruise Control.
+     *
+     * @param passwordGenerator the password generator used for creating new credentials.
+     * @param oldCruiseControlApiSecret the existing centralized API secret, containing previously stored credentials.
+     * @param userManagedApiSecret the secret managed by the user, containing user-defined API credentials.
+     * @param topicOperatorManagedApiSecret the secret managed by the topic operator, containing credentials for the topic operator.
+     * @return a new Secret object containing the aggregated API credentials for Cruise Control.
+     */
+    public Secret generateApiSecret(PasswordGenerator passwordGenerator,
+                                    Secret oldCruiseControlApiSecret,
+                                    Secret userManagedApiSecret,
```

Comment:
Maybe I missed it, but do you have some validation what if the user configured to use a Secret but that Secret is missing? We should throw there some InvalidResourceException or something.

Reply:
Nice catch, added